src/index.js

```diff
@@ -28,9 +28,14 @@ const getUrlString = url => {
   }
 };
 
-export default async ({ cache, markdownAST }) => {
+export default async (
+  { cache, markdownAST },
+  { customTransformers = [] } = {}
+) => {
   const transformations = [];
 
+  const transformers = defaultTransformers.concat(customTransformers);
```

Comment:
Maybe inline this, instead of creating an unnecessary variable? 🤔

Reply:
I didn't inline it since it will mean that the `concat` call will be executed in every node visit...

Reply:
Sounds logical indeed 🙂

Maybe make this the first thing that happens before initializing `transformations` tho? 🤔

Reply:
What about declaring the built-in transformers within the plugin function, and using the spread syntax to add the `customTransformers`?:

```js
// ...

export default async (
  { cache, markdownAST },
  { customTransformers = [] } = {}
) => {
  const transformations = [];

  const transformers = [
    CodePenTransformer,
    CodeSandboxTransformer,
    SlidesTransformer,
    SpotifyTransformer,
    SoundCloudTransformer,
    TwitterTransformer,
    YouTubeTransformer,
    ...customTransformers,
  ];

// ...
```

Reply:
I like the built in ones to be at the top of the file, so they don't really pollute the plugin's code

Reply:
I see what you mean. I was thinking, to avoid polluting the core plugin's code, wouldn't you like to have the built-in transformers to be implemented in a separate file?. For instance, having all built-in transformer modules in a separate directory, and exporting them from `src/transformers/index.js`?

I think it could make sense. If you are adding a new built-in transformer you don't need to change the plugin's core at all, encouraging the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle).

Now coming back to this PR, there we could export a function that receives the `options` argument and adds the custom ones to the array, something along the lines:

```js
// src/transformers/index.js
import * as CodePenTransformer from './CodePen';
import * as CodeSandboxTransformer from './CodeSandbox';
import * as SlidesTransformer from './Slides';
import * as SoundCloudTransformer from './SoundCloud';
import * as SpotifyTransformer from './Spotify';
import * as TwitterTransformer from './Twitter';
import * as YouTubeTransformer from './YouTube';

export function getTransformers({ customTransformers = [] }) {
  return [
    CodePenTransformer,
    CodeSandboxTransformer,
    SlidesTransformer,
    SpotifyTransformer,
    SoundCloudTransformer,
    TwitterTransformer,
    YouTubeTransformer,
    ...customTransformers,
  ];
}
```

Reply:
I think this is indeed the way to go eventually, but for now let's just keep this PR as simple/clean as possible without doing to much

Reply:
Sure thing, I will make a separate PR for that after this one is merged.

Reply:
Sounds like a better plan to me to do it in a separate PR indeed 🙂